include/agent.h

```diff
@@ -0,0 +1,96 @@
+#ifndef _ABM_AGENT_H_
+#define _ABM_AGENT_H_
+
+#include <memory>
+#include <vector>
+
+#include "config.h"
+#include "graph.h"
+#include "agent.h"
+
+namespace abm {
+
+//! Agent class that has id, origin, destination, departure time, current node and compute path from current node to destination.
+class Agent {
+ public:
+  //! Construct an agent class
+  //! \param[in] graph Pass a reference of the graph that the agent does routing on
+  explicit Agent(const std::shared_ptr<abm::Graph>& graph) : graph_{graph} {};
+
+  //! Set id 
+  //! \param[in] agent_id Agent id assigned to the agent object
+  void set_id(graph::vertex_t agent_id) { this->agent_id_ = agent_id; }
```

Comment:
This should be done at constructor level

Reply:
Does this mean the `id` should be passed to the constructor?